File: Android project/RoomDataBase/app/src/main/java/com/mirkamol/roomdatabase/database/UserDao.java
package com.mirkamol.roomdatabase.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.mirkamol.roomdatabase.model.User;
import java.util.List;
@Dao
public interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long saveUser(User user);

    @Query("select * from user_table ")
    List<User> getUsers();

}

File: Android project/SavedtoRoomfromServer/app/src/main/java/com/mirkamol/savedtoroomfromserver/database/UserDao.kt
package com.mirkamol.savedtoroomfromserver.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.mirkamol.savedtoroomfromserver.model.UserItem

@Dao
interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveUser(userDB: UserItem)

    @Query("SELECT *FROM user")
    fun getUsers():List<UserItem>

    @Query("DELETE FROM user")
    fun deleteAllUsers()
}